contracts/loan_manager/src/contract.rs

```diff
@@ -378,6 +390,77 @@ impl LoanManager {
     pub fn get_interest_rate(e: Env, pool: Address) -> i128 {
         get_interest(e, pool)
     }
+
+    pub fn liquidate(e: Env, user: Address, borrower: Address, amount: i128) -> (i128, i128, i128) {
+        user.require_auth();
+
+        let Loan {
+            borrower,
+            borrowed_amount,
+            borrowed_from,
+            collateral_from,
+            collateral_amount,
+            health_factor,
+            unpaid_interest,
+        } = Self::get_loan(&e, borrower);
+
+        let key = (Symbol::new(&e, "Loan"), borrower.clone());
+
+        let borrow_pool_client = loan_pool::Client::new(&e, &borrowed_from);
+        let collateral_pool_client = loan_pool::Client::new(&e, &collateral_from);
+
+        let borrowed_ticker = borrow_pool_client.get_currency().ticker;
+        let collateral_ticker = collateral_pool_client.get_currency().ticker;
+
+        let borrowed_price = Self::get_price(&e, borrowed_ticker.clone());
+        let collateral_price = Self::get_price(&e, collateral_ticker.clone());
+
+        const TEMP_BONUS: i128 = 10_500_000; // multiplier 1.05 -> 5%
+
+        let liquidation_value = amount * borrowed_price;
+        let collateral_amount_bonus =
+            (liquidation_value * TEMP_BONUS / collateral_price) / 10_000_000;
+
+        assert!(health_factor < 12000000); // Temp high value for testing
+        assert!(amount < (borrowed_amount / 2));
+
+        borrow_pool_client.liquidate(&user, &amount, &unpaid_interest, &borrower);
+
+        collateral_pool_client.liquidate_transfer_collateral(
+            &user,
+            &collateral_amount_bonus,
+            &borrower,
+        );
+
+        let new_borrowed_amount = borrowed_amount - amount;
+        let new_collateral_amount = collateral_amount - collateral_amount_bonus;
+
+        let new_health_factor = Self::calculate_health_factor(
+            &e,
+            borrowed_ticker,
+            new_borrowed_amount,
+            collateral_ticker,
+            new_collateral_amount,
+        );
+
+        let new_loan = Loan {
+            borrower,
+            borrowed_amount: new_borrowed_amount,
+            borrowed_from,
+            collateral_from,
+            collateral_amount: collateral_amount - collateral_amount_bonus,
```

Comment:
You already calculted this in the `new_collateral_amount` variable, accident?

Reply:
Yes that's a mistake. I'll fix it!